frontend/src/static/js/main.js

```diff
@@ -159,7 +159,7 @@ window.app = new window.Vue({
       };
       const tabInfoLength = Object.values(zoomInfo).filter((x) => x !== null).length;
       if (Object.keys(zoomInfo).length === tabInfoLength) {
-        this.$store.commit('updateTabZoomInfo', zoomInfo);
+        this.$refs.summary.restoreZoomFiltered(zoomInfo);
```

Comment:
We tried to avoid use this.$ref/$parent etc, could you try to refactor this part since you are on the way projectify the app?

Reply:
I can't think of any easy way to do this.

Reply:
Actually this can be done by moving some more code out of summary.js

Reply:
Since you actually have another PR for modifying summary.js, you can do the changes needed there and we will merge this PR later.

Reply:
Actually there's data stored in `summary.js` which cannot really be moved out unless you move out a lot from `summary.js`. I can try to remove `this.ref` after #1418 is merged however.

Reply:
Merged #1418